posthog/auth.py

```diff
@@ -94,7 +94,12 @@ def authenticate(cls, request: Union[HttpRequest, Request]) -> Optional[Tuple[An
         assert personal_api_key_object.user is not None
 
         # :KLUDGE: CHMiddleware does not receive the correct user when authenticating by api key.
-        tag_queries(user_id=personal_api_key_object.user.pk, team_id=personal_api_key_object.user.current_team_id)
+        tag_queries(
+            user_id=personal_api_key_object.user.pk,
+            team_id=personal_api_key_object.user.current_team_id,
+            access_method="personal_api_key",
+        )
+        request.using_personal_api_key = True  # type: ignore
```

Comment:
This flag is used elsewhere?

Reply:
Will be in the linked PR. Going to be lazy to keep rebasing easy.